src/lib.rs

```diff
@@ -283,6 +286,47 @@ impl<C> PipelineBuilder<C> {
     ///     .install_simple();
     /// ```
     pub fn with_trace_config(self, config: sdk::trace::Config) -> Self {
+        let merged_resource = match self.config {
+            Some(base_config) => base_config.resource.merge(&config.resource),
+            None => (*config.resource).clone(),
+        };
+
+        let config = config.with_resource(merged_resource);
+
+        PipelineBuilder {
+            config: Some(config),
+            ..self
+        }
+    }
+
+    /// Assign the service name under which to group traces by adding a service.name
+    /// `sdk::Resource` or overriding a previous setting of it.
```

Comment:
Please note that "overriding a previous setting" is not true with the current version 0.13.0 of `opentelemetry`. We just updated the `sdk::Resource::merge` logic to reflect the latest spec in https://github.com/open-telemetry/opentelemetry-rust/pull/537. This will be included in the next release, though.